### PR TITLE
[WC-2046]: Implement Gallery Keyboard Navigation

### PR DESCRIPTION
**Pull request by samuelreichert**
### Pull request type

New feature (non-breaking change which adds functionality)

---

### CODE REVIEW

File: packages/pluggableWidgets/gallery-web/src/Gallery.editorPreview.tsx [hunk 1]
```diff
@@ -13,7 +16,27 @@ function Preview(props: GalleryPreviewProps): ReactElement {
     }));
 
     const selectHelper = useItemSelectHelper(props.itemSelection, undefined);
-    const itemEventsController = useItemEventsController(selectHelper);
+
+    const { numberOfColumns, numberOfRows } = useGridPositions({
+        phoneItems: props.phoneItems ?? 1,
+        tabletItems: props.tabletItems ?? 2,
+        desktopItems: props.desktopItems ?? 3,
```
Comment (iobuhov) [inline packages/pluggableWidgets/gallery-web/src/Gallery.editorPreview.tsx:23]: Let use `1` as a fallback for each breakpoint

File: packages/pluggableWidgets/gallery-web/src/features/item-interaction/__tests__/item-keyboard.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    //             await user.tab();
+    //             await user.keyboard(`{Shift>}[${key}]{/Shift}`);
+
+    //             expect(onSelectAdjacent).toHaveBeenCalledTimes(n);
+    //             if (n > 0) {
+    //                 expect(onSelectAdjacent).toHaveBeenLastCalledWith(item, ...params);
+    //             }
+    //         }
+    //     );
+    // });
```
Comment (iobuhov) [inline packages/pluggableWidgets/gallery-web/src/features/item-interaction/__tests__/item-keyboard.spec.tsx:117]: Let remove this test for now?

File: packages/pluggableWidgets/gallery-web/src/features/use-grid-positions.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import { ObjectItem } from "mendix";
```
Comment (iobuhov) [inline packages/pluggableWidgets/gallery-web/src/features/use-grid-positions.ts:1]: Let rename this file (and others) using camelCase.

File: packages/pluggableWidgets/gallery-web/src/features/item-interaction/use-item-events-controller.ts [hunk 4]
```diff
@@ -1,15 +1,23 @@
 import { useMemo } from "react";
```
Comment (iobuhov) [inline packages/pluggableWidgets/gallery-web/src/features/item-interaction/use-item-events-controller.ts:1]: Move this hook to ItemEventsController

File: packages/pluggableWidgets/gallery-web/src/helpers/use-click-action-helper.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { useEffect, useMemo } from "react";
```
Comment (iobuhov) [inline packages/pluggableWidgets/gallery-web/src/helpers/use-click-action-helper.ts:1]: Move this hook to ClickActionHelper.ts

**Review by iobuhov — CHANGES_REQUESTED** [review]

File: packages/shared/tsconfig-web-widgets/esm-library-with-jsx.json [hunk 6]
```diff
@@ -4,9 +4,9 @@
         "declarationMap": true,
         "noEmitOnError": true,
         "sourceMap": true,
-        "module": "ES2022",
+        "module": "esnext",
         "target": "ES2021",
-        "lib": ["ES2021", "dom"],
+        "lib": ["esnext", "dom"],
```
Comment (iobuhov) [inline packages/shared/tsconfig-web-widgets/esm-library-with-jsx.json:9]: Let's replace `esnext` with `ES2022`

**Review by iobuhov — CHANGES_REQUESTED** [review]

File: packages/shared/tsconfig-web-widgets/esm-library-with-jsx.json (hunk 6 above)
Comment (iobuhov) [reply packages/shared/tsconfig-web-widgets/esm-library-with-jsx.json:9]: + Let's update `target` as well

**Review by iobuhov — APPROVED** [review]
LGTM

**Review by iobuhov — APPROVED** [review]
LGTM

**Review by leonardomendix — DISMISSED** [review]

**Review by leonardomendix — APPROVED** [review]

**Review by iobuhov — APPROVED** [review]

**Pull request merged**